### PR TITLE
Check passes length in Shadow.getPlanarShader & getPlanarInstanceShader.

### PR DESCRIPTION
**Pull request by dumganhar**
If developer doesn't check `Project Settings -> Feature Cropping -> 3D -> Basic 3D Features
`  for a 2D game.
, game will crash on native platforms or throw exception on web.

This PR make sure there are not crashes or exceptions.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/scene/Shadow.cpp [hunk 1]
```diff
@@ -138,17 +138,17 @@ gfx::Shader *Shadows::getPlanarShader(const ccstd::vector<IMacroPatch> &patches)
         createMaterial();
     }
 
-    const auto &passes = *_material->getPasses();
-    return passes[0]->getShaderVariant(patches);
+    const auto &passes = _material->getPasses();
+    return passes && !passes->empty() ? passes->at(0)->getShaderVariant(patches) : nullptr;
```
Comment (minggo) [inline native/cocos/scene/Shadow.cpp]: Why need to check `passes` first?

Comment (dumganhar) [reply native/cocos/scene/Shadow.cpp]: Because it's a std::shared_ptr.

**Comment by dumganhar** [comment]
@cocos-robot run test cases

File: native/cocos/scene/Shadow.cpp (hunk 1 above)
Comment (minggo) [reply native/cocos/scene/Shadow.cpp]: Got it.

**Review by minggo — APPROVED** [review]

**Review by troublemaker52025 — APPROVED** [review]

**Comment by dumganhar** [comment]
@cocos-robot run test cases

**Review by SantyWang — APPROVED** [review]

**Pull request merged**